tests/test_zuora_automatic_fields.py

```diff
@@ -0,0 +1,73 @@
+from tap_tester import runner, connections
+from base import ZuoraBaseTest
+
+
+class ZuoraAutomaticFields(ZuoraBaseTest):
+    """Test that with no fields selected for a stream automatic fields are still replicated"""
+    @staticmethod
+    def name():
+        return "tap_tester_zuora_automatic_fields_rest"
+
+    def test_run(self):
+        """ Executing tap-tester scenarios for both types of zuora APIs AQUA and REST"""
+        self.run_test("AQUA")
+        # self.run_test("REST")
+
+    def run_test(self, api_type):
+        """
+        Verify that for each stream you can get multiple pages of data
+        when no fields are selected and only the automatic fields are replicated.
+
+        PREREQUISITE
+        For EACH stream add enough data that you surpass the limit of a single
+        fetch of data.  For instance if you have a limit of 250 records ensure
+        that 251 (or more) records have been posted for that stream.
+        """
+        self.zuora_api_type = api_type
+        expected_streams =  {'Export', 'RatePlan', 'Account'}
+
+        # Instantiate connection
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        # Run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Table and field selection
+        test_catalogs_automatic_fields = [catalog for catalog in found_catalogs
+                                          if catalog.get('stream_name') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs_automatic_fields, select_all_fields=False,)
+
+        # Run initial sync
+        record_count_by_stream = self.run_and_verify_sync(conn_id)
+        synced_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # Expected values
+                expected_keys = self.expected_automatic_fields().get(stream)
+
+
+                expected_primary_keys = self.expected_primary_keys()[stream]
+
+                # Collect actual values
+                data = synced_records.get(stream, {})
+                record_messages_keys = [set(row.get('data').keys()) for row in data.get('messages', {})]
+                primary_keys_list = [tuple(message.get('data', {}).get(expected_pk) for expected_pk in expected_primary_keys)
+                                    for message in data.get('messages', [])
+                                    if message.get('action') == 'upsert']
+                unique_primary_keys_list = set(primary_keys_list)
+
+                # Verify that you get some records for each stream
+                self.assertGreater(
+                    record_count_by_stream.get(stream, -1), 0,
+                    msg="The number of records is not over the stream max limit for the {} stream".format(stream))
```

Comment:
We don't need this assert explicitly since it is taken care by `record_count_by_stream = self.run_and_verify_sync(conn_id)` here https://github.com/singer-io/tap-zuora/blob/TDL-20774-adding-integration-tests/tests/base.py#L350

Reply:
Made the changes